crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -3280,6 +3285,21 @@ pub(crate) mod tests {
         .await
         .unwrap();
 
+        // We should receive a notification on the room_keys_withheld_received_stream
+        let withheld_received = room_keys_withheld_received_stream
+            .next()
+            .now_or_never()
```

Comment:
To use `now_or_never`, you must be sure that the value will be immediately available in the `Stream`, otherwise the test will be flaky.

Is it a problem to use `next().await` here instead? The risk is that the test might be pending forever if something changes in the future, but it's still an indication that a regression has been detected. The pros, however, is that we are sure the test can never be flaky.

Reply:
> Is it a problem to use `next().await` here instead?

I guess not. TBH I pretty much cargo-culted this from `test_megolm_encryption` above, which afaik has never shown any sign of being flaky. Flipping it around: is there any reason to believe that the value would *not* already be available in the Stream?

Reply:
The stream is created before we receive the sync changes, the sync changes trigger an update, this means that the `now_or_never()` should never fail here, unless we introduce a bug.

There are no hidden tasks/threads or requests being sent out in `receive_sync_changes()` it's all a pretty dumb, for better or worse, state machine.

`now_or_never()` is the right thing to use, otherwise a regression will produce a hang instead of a loud test failure.